docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst

```diff
@@ -0,0 +1,618 @@
++---------------------------------+--------------+----------------+---------+
+| ``begin`` StmtSequence_ ``end`` | BeginExpr_   | StmtSequence_  |         |
++---------------------------------+--------------+----------------+---------+
+| ``break`` «Expr_»?              | BreakStmt_   | ReturningStmt_ |         |
++---------------------------------+--------------+----------------+---------+
+| ``;``                           | EmptyStmt_   | Stmt_          |         |
++---------------------------------+--------------+----------------+---------+
+| ``END {`` StmtSequence_ ``}``   | EndBlock_    | StmtSequence_  |         |
++---------------------------------+--------------+----------------+---------+
+| ``next`` «Expr_»?               | NextStmt_    | ReturningStmt_ |         |
++---------------------------------+--------------+----------------+---------+
+| ``redo``                        | RedoStmt_    | Stmt_          |         |
++---------------------------------+--------------+----------------+---------+
+| ``retry``                       | RetryStmt_   | Stmt_          |         |
++---------------------------------+--------------+----------------+---------+
+| ``return`` «Expr_»?             | ReturnStmt_  | ReturningStmt_ |         |
++---------------------------------+--------------+----------------+---------+
+| ``undef`` «FNAME ``,``»+        | UndefStmt_   | Stmt_          |         |
++---------------------------------+--------------+----------------+---------+
+
+Calls
+~~~~~
+
++----------------------------+---------------------+----------------+-------------------------------+
+| Expression syntax          | CodeQL class        | Superclasses   | Remarks                       |
++============================+=====================+================+===============================+
+| Expr_ ``[`` Expr_ ``]``    | ElementReference_   | MethodCall_    |                               |
++----------------------------+---------------------+----------------+-------------------------------+
+| MethodName_ «Expr_ ``,``»* | MethodCall_         | Call_          |                               |
++----------------------------+---------------------+----------------+-------------------------------+
+| LhsExpr_ ``=`` Expr_       | SetterMethodCall_   | MethodCall_    |                               |
++----------------------------+---------------------+----------------+-------------------------------+
+| ``super``                  | SuperCall_          | MethodCall_    |                               |
++----------------------------+---------------------+----------------+-------------------------------+
+| ``yield`` «Expr_ ``,``»*   | YieldCall_          | Call_          |                               |
++----------------------------+---------------------+----------------+-------------------------------+
+| ``&``IDENTIFIER            | BlockArgument_      | Expr_          | Used as an argument to a call |
```

Comment:
Sphinx says: `Inline literal start-string without end-string.` . Perhaps we need a space between `&` and _IDENTIFIER_

See: https://github.com/github/semmle-code/actions/runs/3395113988/jobs/5644570096#step:18:17

Reply:
You may well be right. I assumed that it would be okay because line 63 appeared to have a double-backtick followed by another symbol without a space.

Reply:
Hopefully fixed now.